sql/moz-fx-data-shared-prod/telemetry_derived/investigation_aggregates_v1/query.sql

```diff
@@ -0,0 +1,256 @@
+    DATE(submission_timestamp) >= '2021-01-01'
+  GROUP BY
+    1
+  UNION ALL
+  -- Fenix nightly
+  SELECT
+    client_info.client_id,
+    MIN(DATE(submission_timestamp)) AS submission_date,
+    CASE
+    WHEN
+      ARRAY_AGG(metrics.string.metrics_adjust_network)[SAFE_OFFSET(0)] NOT IN (
+        '',
+        'Organic',
+        'Google Organic Search',
+        'Untrusted Devices',
+        'Product Marketing (Owned media)',
+        'Google Ads ACI'
+      )
+      AND ARRAY_AGG(metrics.string.metrics_adjust_network)[SAFE_OFFSET(0)] IS NOT NULL
+    THEN
+      'Other'
+    ELSE
+      ARRAY_AGG(metrics.string.metrics_adjust_network)[SAFE_OFFSET(0)]
+    END
+    AS adjust_network,
+    CASE
+    WHEN
+      ARRAY_AGG(metrics.string.metrics_install_source)[SAFE_OFFSET(0)] NOT IN (
+        'com.android.vending'
+      )
+      AND ARRAY_AGG(metrics.string.metrics_install_source)[SAFE_OFFSET(0)] IS NOT NULL
+    THEN
+      'Other'
+    ELSE
+      ARRAY_AGG(metrics.string.metrics_install_source)[SAFE_OFFSET(0)]
+    END
+    AS install_source
+  FROM
+    org_mozilla_fennec_aurora.metrics AS org_mozilla_fennec_aurora_metrics
+  WHERE
+    DATE(submission_timestamp) >= '2021-01-01'
+  GROUP BY
+    1
+  UNION ALL
+  -- Fenix beta
+  SELECT
+    client_info.client_id,
+    MIN(DATE(submission_timestamp)) AS submission_date,
+    CASE
+    WHEN
+      ARRAY_AGG(metrics.string.metrics_adjust_network)[SAFE_OFFSET(0)] NOT IN (
+        '',
+        'Organic',
+        'Google Organic Search',
+        'Untrusted Devices',
+        'Product Marketing (Owned media)',
+        'Google Ads ACI'
+      )
+      AND ARRAY_AGG(metrics.string.metrics_adjust_network)[SAFE_OFFSET(0)] IS NOT NULL
+    THEN
+      'Other'
+    ELSE
+      ARRAY_AGG(metrics.string.metrics_adjust_network)[SAFE_OFFSET(0)]
+    END
+    AS adjust_network,
+    CASE
+    WHEN
+      ARRAY_AGG(metrics.string.metrics_install_source)[SAFE_OFFSET(0)] NOT IN (
+        'com.android.vending'
+      )
+      AND ARRAY_AGG(metrics.string.metrics_install_source)[SAFE_OFFSET(0)] IS NOT NULL
+    THEN
+      'Other'
+    ELSE
+      ARRAY_AGG(metrics.string.metrics_install_source)[SAFE_OFFSET(0)]
+    END
+    AS install_source
+  FROM
+    org_mozilla_firefox_beta.metrics AS org_mozilla_firefox_beta_metrics
+  WHERE
+    DATE(submission_timestamp) >= '2021-01-01'
+  GROUP BY
+    1
+  UNION ALL
+  -- Fenix release
+  SELECT
+    client_info.client_id,
+    MIN(DATE(submission_timestamp)) AS submission_date,
+    CASE
+    WHEN
+      ARRAY_AGG(metrics.string.metrics_adjust_network)[SAFE_OFFSET(0)] NOT IN (
+        '',
+        'Organic',
+        'Google Organic Search',
+        'Untrusted Devices',
+        'Product Marketing (Owned media)',
+        'Google Ads ACI'
+      )
+      AND ARRAY_AGG(metrics.string.metrics_adjust_network)[SAFE_OFFSET(0)] IS NOT NULL
+    THEN
+      'Other'
+    ELSE
+      ARRAY_AGG(metrics.string.metrics_adjust_network)[SAFE_OFFSET(0)]
+    END
```

Comment:
Maybe to reduce the verbosity we could move this logic into a UDF? Seems like the logic is used multiple times?